diff -ruN --no-dereference dolfinx-0.3.0.orig/python/dolfinx/__init__.py dolfinx-0.3.0/python/dolfinx/__init__.py
--- dolfinx-0.3.0.orig/python/dolfinx/__init__.py	2022-03-19 10:42:33.544544141 +0800
+++ dolfinx-0.3.0/python/dolfinx/__init__.py	2022-03-19 10:43:22.157918727 +0800
@@ -7,22 +7,22 @@
 
 # flake8: noqa
 
-# Store dl open flags to restore them after import
-import sys
-stored_dlopen_flags = sys.getdlopenflags()
-
-# Developer note: below is related to OpenMPI
-# Fix dlopen flags (may need reorganising)
-if "linux" in sys.platform:
-    # FIXME: What with other platforms?
-    try:
-        from ctypes import RTLD_NOW, RTLD_GLOBAL
-    except ImportError:
-        RTLD_NOW = 2
-        RTLD_GLOBAL = 256
-    sys.setdlopenflags(RTLD_NOW | RTLD_GLOBAL)
-del sys
+## Store dl open flags to restore them after import
+#import sys
+#stored_dlopen_flags = sys.getdlopenflags()
 
+## Developer note: below is related to OpenMPI
+## Fix dlopen flags (may need reorganising)
+#if "linux" in sys.platform:
+#    # FIXME: What with other platforms?
+#    try:
+#        from ctypes import RTLD_NOW, RTLD_GLOBAL
+#    except ImportError:
+#        RTLD_NOW = 2
+#        RTLD_GLOBAL = 256
+#    sys.setdlopenflags(RTLD_NOW | RTLD_GLOBAL)
+#del sys
+############################################################
 # Reset dl open flags
 # sys.setdlopenflags(stored_dlopen_flags)
 # del sys
