diff -ruN jupyterlab-2.1.1.orig/jupyterlab/commands.py jupyterlab-2.1.1/jupyterlab/commands.py
--- jupyterlab-2.1.1.orig/jupyterlab/commands.py	2020-04-26 13:20:20.481196574 +0000
+++ jupyterlab-2.1.1/jupyterlab/commands.py	2020-04-26 13:21:48.609357717 +0000
@@ -147,7 +147,7 @@
         return osp.abspath(os.environ['JUPYTERLAB_DIR'])
 
     # Use the default locations for data_files.
-    app_dir = pjoin(sys.prefix, 'share', 'jupyter', 'lab')
+    app_dir = pjoin(os.environ.get('EBROOTJUPYTERLAB',sys.prefix), 'share', 'jupyter', 'lab')
 
     # Check for a user level install.
     # Ensure that USER_BASE is defined
