diff -ru installer.orig/Makefile.ver installer/Makefile.ver
--- installer.orig/Makefile.ver	2019-10-20 15:36:47.000000000 +0800
+++ installer/Makefile.ver	2019-10-24 06:53:20.181346633 +0800
@@ -132,11 +132,17 @@
 clean: $(addprefix CLEAN_,$(SWORDS))
 cleaner: $(addprefix CLEANER_,$(SWORDS))
 cleanall:
-	git clean -n -f -x -d
+	$(if $(GIT_OFFLINE),\
+        $(warning Skipping cleanall (GIT_OFFLINE is nonempty)),\
+		{ git clean -n -f -x -d }
+	)
 	@echo Purposefully not cleaning. Run \'git clean -i -x -d\' yourself for interactive cleaning.
 version: $(addprefix VER_,$(SWORDS))
 	@echo "SITECONFIG: $(SITECONFIG)."
-	@echo Installer version: $(shell $(GITD) --dirty)
+	$(if $(GIT_OFFLINE),\
+        {@echo Installer version: ${VER_DD}_${VER_AL}},\
+		{@echo Installer version: $(shell $(GITD) --dirty) }
+	)	
 	@echo MODULE_FULL_VERSION: $(MODULE_FULL_VERSION)
 	$(if $(VERBOSE),@echo "IMAS_MAJOR: $(IMAS_MAJOR).",)
 	$(if $(VERBOSE),@echo "IMAS_MINOR: $(IMAS_MINOR).",)
@@ -174,11 +180,14 @@
 
 # Checkout and pull if COMMIT or BRANCH is set
 $(addprefix CHECKOUT_,$(SWORDS)): CHECKOUT_%:CACHE_%
-	$(if $(wildcard $($@)/.git),\
-		{ cd $($@) && git fetch --tags && git checkout -q $($(subst CHECKOUT,CB,$@)) $(if $($(subst CHECKOUT,COMMIT,$@)),,&& git pull --ff-only) ;},\
-		{ git clone $($(subst CHECKOUT,CACHE,$@)) $($@) && cd $($@) && git checkout -q $($(subst CHECKOUT,CB,$@)) ;})
-	$(if $(VERBOSE),{ cd $($@) && git status ;},)
- 
+	$(if $(GIT_OFFLINE),\
+        $(warning Skipping CHECKOUT (GIT_OFFLINE is nonempty)),\
+		$(if $(wildcard $($@)/.git),\
+			{ cd $($@) && git fetch --tags && git checkout -q $($(subst CHECKOUT,CB,$@)) $(if $($(subst CHECKOUT,COMMIT,$@)),,&& git pull --ff-only) ;},\
+			{ git clone $($(subst CHECKOUT,CACHE,$@)) $($@) && cd $($@) && git checkout -q $($(subst CHECKOUT,CB,$@)) ;}) \
+	)
+	$(if $(GIT_OFFLINE),	$(if $(VERBOSE),{ cd $($@) && git status ;},),)
+
 # Create a clone mirror as cache or update mirror
 $(addprefix CACHE_,$(SWORDS)):
 	$(if $(GIT_OFFLINE),\
