diff --git a/libs/MUMPS_4.7.3/MATLAB/Makefile b/libs/MUMPS_4.7.3/MATLAB/Makefile
index cecc399..b079783 100755
--- a/libs/MUMPS_4.7.3/MATLAB/Makefile
+++ b/libs/MUMPS_4.7.3/MATLAB/Makefile
@@ -19,10 +19,10 @@ ifeq ($(MATLABVERSION),-DMATLAB86)
 	$(MEX_F90) -largeArrayDims -fortran $(OPTC)  dmumpsmex.o $(INCTOT) $(LIBTOT)
 else ifeq ($(MATLABVERSION),-DMATLAB73)
 	$(MEX) $(OPTC) -largeArrayDims -c $(INCTOT) -I$(MATLABHOME)/extern/include dmumpsmex.c 
-	$(MEX_F90) -largeArrayDims -fortran $(OPTC) -o dmumpsmex.$(EXT) dmumpsmex.o $(INCTOT) $(LIBTOT)
+	$(MEX_F90) -largeArrayDims -fortran $(OPTC) -output dmumpsmex.$(EXT) dmumpsmex.o $(INCTOT) $(LIBTOT)
 else
 	$(MEX) $(OPTC) -c $(INCTOT) -I$(MATLABHOME)/extern/include dmumpsmex.c
-	$(MEX_F90) -fortran $(OPTC) -o dmumpsmex.$(EXT) dmumpsmex.o $(INCTOT) $(LIBTOT)
+	$(MEX_F90) -fortran $(OPTC) -output dmumpsmex.$(EXT) dmumpsmex.o $(INCTOT) $(LIBTOT)
 endif
 clean:
 	rm -f *.o *.$(EXT)
diff --git a/solver/Makefile b/solver/Makefile
index ca7b267..6dc1c8b 100755
--- a/solver/Makefile
+++ b/solver/Makefile
@@ -17,7 +17,7 @@ interpos_old.$(EXT): interpos_old.f90 $(includefile)
 	$(MEX) $<
 	
 interpos.$(EXT): interpos.f90
-	mex $<
+	$(MEX) $<
 
 pde1solver.o: pde1solver.f90
 	$(F90) -c $(FFLAGS_MUMPS) -I$(MUMPSHOME)/libseq -I$(MUMPSHOME)/include $<
