diff -ruN --no-dereference pdsh-pdsh-2.34.orig/configure.ac pdsh-pdsh-2.34/configure.ac
--- pdsh-pdsh-2.34.orig/configure.ac	2022-03-24 17:52:48.019640746 +0800
+++ pdsh-pdsh-2.34/configure.ac	2022-03-24 17:53:11.764857810 +0800
@@ -6,8 +6,8 @@
 #
 # This file is to be processed with autoconf to generate a configure script.
 
-AC_INIT([pdsh],
-        m4_esyscmd([git describe --always |  awk '/.*/ {sub(/^pdsh-/, ""); printf "%s",$1; exit}']))
+AC_INIT([pdsh], 2.34 )
+ #       m4_esyscmd([git describe --always |  awk '/.*/ {sub(/^pdsh-/, ""); printf "%s",$1; exit}']))
 
 AC_CONFIG_SRCDIR([pdsh])
 AC_CONFIG_AUX_DIR([config])
